docs/functional/architecture/0004-Lifted_Stencils_with_Tuple_Return.md

```diff
@@ -0,0 +1,88 @@
+---
+tags: [iterator]
+---
+
+# Lifted Stencils with Tuple Return
+
+- **Status**: valid
+- **Authors**: Hannes Vogt (@havogt)
+- **Created**: 2022-02-16
+- **Updated**: 2022-02-16
+
+What's the return type of a lifted stencil that has tuple return type? Is it tuple of iterator or iterator of tuple?
+
+
+## Context
+
+Example
+
+```python
+def stencil():
+    return 42, 43
+
+# What is the type of
+lift(stencil)()
+```
```

Comment:
I am not super up-to-date how the current lowering is implemented, but I see the following way:

from field view
```python
def fieldop(a: Field[...]):
    return a + 42, a + 43
```
to iterator ir:
```python
def fieldop(a_it):
    return make_tuple(lift(lambda a: deref(a)+42)(a), lift(lambda a: deref(a)+43)(a))
```
Note that this is not a stencil as it does not return value. It's also not a lifted stencil because it returns tuple of iterator. But it is a legal function in iterator IR.

Does it make sense?

Reply:
Looks reasonable and also looks like the details of how to use this in the field view wrappers will be easy to work out.